### PR TITLE
Add Association Feature

### PR DESCRIPTION
**Pull request by bushmusi**
> # Below features have been added

- [x] Set up associations between classes and objects
- [x] Add `classroom`, `rental`, and `book` file

[**Project requirement**](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_associations.md)

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+  def play_hooky
+    "¯\(ツ)/¯"
+  end
+
+  def classroom=(new_classroom)
+    @classroom.remove_student(self) unless @classroom.nil? || @classroom == new_classroom
+
+    @classroom = new_classroom
+    @classroom.add_student(self) unless @classroom.students.include?(self)
```
Comment (DeliceLydia) [inline student.rb:19]: - [ ] This is a `belongs_to` side(a student belongs to a classroom) you can use `students.push(self)`:
`classroom.students.push(self) unless classroom.students.include?(self)`

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+require './rental'
+
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
```
Comment (DeliceLydia) [inline book.rb]: - [ ] I suggest that you use `rentals` as `attr_accessor` as well

File: rental.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+class Rental
+  attr_reader :person, :book
```
Comment (DeliceLydia) [inline rental.rb]: - [ ] I suggest that you use `person` and `book` as `attr_accessor` to be able to access and modify them at the same time

File: person.rb [hunk 4]
```diff
@@ -0,0 +1,36 @@
+require './nameable'
+
+class Person < Nameable
+  attr_accessor :name, :age
+
+  attr_reader :id, :rentals
```
Comment (DeliceLydia) [inline person.rb]: - [ ] Please make `rentals` as `attr_accessor` as well to be able to access them and modify them at the same time

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi @bushmusi, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there! Highlights for your project:
- Kudos on the good PR
- Followed git workflow
- No linter errors

## Required Changes ♻️
_Check the comments under the review._



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)

File: book.rb (hunk 2 above)
Comment (bushmusi) [reply book.rb]: Understood and improved

File: person.rb (hunk 4 above)
Comment (bushmusi) [reply person.rb]: Thanks and solved

File: rental.rb (hunk 3 above)
Comment (bushmusi) [reply rental.rb]: Thanks and solved

**Review by GzimAsani — CHANGES_REQUESTED** [review]
Hi @bushmusi , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that has linter errors

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: student.rb (hunk 1 above)
Comment (Whoistolu) [reply student.rb:19]: Well resolved 👍🏽

File: book.rb (hunk 2 above)
Comment (Whoistolu) [reply book.rb]: Well resolved 👍🏽

File: rental.rb (hunk 3 above)
Comment (Whoistolu) [reply rental.rb]: Well resolved 👍🏽

File: person.rb (hunk 4 above)
Comment (Whoistolu) [reply person.rb]: Well resolved 👍🏽

**Review by Whoistolu — APPROVED** [review]
### Hi @bushmusi 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by bushmusi** [comment]
> ### Hi @bushmusi
> Your project is complete! There is nothing else to say other than... it's time to merge it :shipit:
> 
> # STATUS: APPROVED 
> **Congratulations! tada**
> 
> Cheers and Happy coding!clapclapclap
> 
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me @Whoistolu in your question so I can receive the notification.
> 
> _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Thanks @Whoistolu

**Pull request merged**